draft-ietf-quic-multipath.md

```diff
@@ -515,6 +480,42 @@ the server MAY wait for a short, limited time such as one PTO if a path
 probing packet is received on a new path before sending the
 CONNECTION_CLOSE frame.
 
+### Avoiding Spurious Stateless Resets {#spurious-stateless-reset}
+
+The peers that send a PATH_ABANDON frame MUST treat all connection
+identifiers received from the peer for the path ID as immediately
+retired. The Stateless Reset Tokens associated with these connection
+identifiers MUST NOT be used to identify Stateless Reset packets
+per {{Section 10.3 of QUIC-TRANSPORT}}.
+
+Due to packet losses and network delays, packets sent on the path may
+well arrive after the PATH_ABANDON frames have been sent or received.
+If these packets arrive after the connection identifiers sent to the peer
+have been retired, they will not be recognizes as bound for the local
+connection and could trigger the peer to send a Stateless Reset
+packet. The rule to "retain knowledge of connection ID for a short period"
+is intended to reduce the risk of sending such spurious stateless
+packets, but it cannot completely avoid that risk.
+
+The immediate retirement of connection identifiers received for the
+path guarantees that spurious stateless reset packets
+sent by the peer will not cause the closure of the QUIC connection.
+
+### Handling ACK_MP for abandoned paths {#ack-after-abandon}
+
+When an endpoint decides to send a PATH_ABANDON frame, there may
+still be some unacknowledged packets. Some other packets may well
+be in transit, and could be received shortly after sending the
+PATH_ABANDON frame. As specified above, the endpoints SHOULD
+send ACK_MP frames promptly, to avoid unnecessary data
+retransmission after the peer deletes path resources.
+
+These ACK_MP frames SHOULD be send on a different path than the
```

Comment:
I think this is a MUST. Or no normatively language here at all. I would rather say somewhere else that if the PATH_ABANDON frame was sent, the endpoint MUST NOT send on that path anymore.

Reply:
Yes, that's probably the best way to handle it, but there is the slight issue of abandoning the last path.

Reply:
For the last path you should do connection close instead. That’s already written somewhere but it might be a SHOULD only. If so, we can maybe change to a MUST.

Reply:
Not sure it matters. Suppose that we treat "abandon of last path" as a protocol error. That means the peer receiving the abandon will automatically issue "connection close" with reason protocol error, and the peer sending the abandon will send nothing, because it must not send data after abandon. But yes, sending connection close is more graceful.

Reply:
Replace SHOULD by MUST.